spyder_kernels/comms/frontendcomm.py

```diff
@@ -127,22 +127,52 @@ def cache_message(self, comm_id, msg):
 
     def notify_comm_ready(self, comm):
         """Send messages about comm readiness to frontend."""
-        self.remote_call(comm.comm_id)._comm_ready()
+        self.remote_call(
+            comm_id=comm.comm_id,
+            callback=self._comm_ready_callback
+        )._comm_ready()
 
         # Cached messages for that comm
         if comm.comm_id in self._cached_messages:
             for msg in self._cached_messages[comm.comm_id]:
                 comm.handle_msg(msg)
             self._cached_messages.pop(comm.comm_id)
 
-    def notify_iopub_ready(self):
-        """Notify frontend that the iopub is ready"""
-        self._iopub_ready = True
-        for comm in self._pending_comms:
-            self.notify_comm_ready(comm)
-        self._pending_comms = []
-
     # --- Private --------
+    def _wait_for_iopub_connect(self, comm):
+        """Wait 10 seconds for iopub new connections."""
```

Comment:
I'm not sure if this docstring is correct because I don't clearly see the 10 seconds in this method's code. But even if this is correct, perhaps we should wait longer on Windows because the kernel usually takes more time to start there (perhaps 20 or 30 secs?).

Reply:
The 10 seconds are in the ` _iopub_monitor_main`
The problem here is that I have two options:
 - I can wait indefinitely, but then the shutdown will be problematic
 - I can have a timeout, but then I need to find a way of handling the case where no messages are received.